src/main/java/gyro/google/compute/SecurityPolicyRule.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2020, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.compute;
+
+import java.util.Set;
+
+import com.google.api.services.compute.Compute;
+import com.google.api.services.compute.model.Operation;
+import gyro.core.GyroUI;
+import gyro.core.resource.Id;
+import gyro.core.resource.Resource;
+import gyro.core.resource.Updatable;
+import gyro.core.scope.State;
+import gyro.core.validation.Required;
+import gyro.core.validation.ValidStrings;
+import gyro.google.Copyable;
+
+public class SecurityPolicyRule extends ComputeResource
+    implements Copyable<com.google.api.services.compute.model.SecurityPolicyRule> {
+
+    private String description;
+    private Integer priority;
+    private String action;
+    private Boolean preview;
+    private SecurityPolicyRuleMatcher match;
+
+    /**
+     * The description of the security policy rule.
+     */
+    @Updatable
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    /**
+     * The priority of the security policy rule.
+     */
+    @Id
```

Comment:
Subresource/complex type should not have the `@Id` annotation, as findById only works on resources.
```suggestion
```